superset/views/core.py

```diff
@@ -1275,13 +1276,44 @@ def filter(self, datasource_type, datasource_id, column):
             return json_error_response(DATASOURCE_MISSING_ERR)
         if not self.datasource_access(datasource):
             return json_error_response(DATASOURCE_ACCESS_ERR)
-
-        payload = json.dumps(
-            datasource.values_for_column(
-                column,
-                config.get('FILTER_SELECT_ROW_LIMIT', 10000),
-            ),
-            default=utils.json_int_dttm_ser)
+        # Implement: Cache endpoint by datasource and column
+        cache_key = hashlib.md5((datasource_id + column).encode('utf-8')).hexdigest()
+        if cache_key and cache:
+            if (hasattr(datasource, 'database') and datasource.database.cache_timeout):
```

Comment:
```
cache_timeout = config.get('CACHE_DEFAULT_TIMEOUT')
try:
   if datasource.database.cache_timeout:
       cache_timeout = ...
except AttributeError:
   if datasource.cache_timeout:
       cache_timeout = ...
```

Reply:
I don't think using try-except is proper here. There are three possible places to get cache_timeout value. I wanted to make it cascading fallback.

Reply:
Updated the code above